nifi-nar-bundles/nifi-morphlines-bundle/nifi-morphlines-processor/src/main/java/org/apache/nifi/processors/morphlines/Morphlines.java

```diff
@@ -0,0 +1,214 @@
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.*;
+import java.util.stream.*;
+import java.util.concurrent.atomic.*;
+import org.apache.nifi.annotation.lifecycle.OnStopped;
+import org.apache.nifi.processor.exception.*;
+
+@Tags({"kitesdk", "morphlines"})
+@CapabilityDescription("Implements Morphlines framework")
+public class MorphlinesProcessor extends AbstractProcessor {
+
+    private static Command morphline;
+    private volatile Record record = new Record();
+    private static volatile Collector collector = new Collector();
+
+    public static final PropertyDescriptor MORPHLINES_ID = new PropertyDescriptor
+            .Builder().name("Morphlines ID")
+            .description("Identifier of the morphlines context")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(true)
+            .build();
+
+    public static final PropertyDescriptor MORPHLINES_FILE = new PropertyDescriptor
+            .Builder().name("Morphlines File")
+            .description("File for the morphlines context")
+            .required(true)
+            .addValidator(StandardValidators.FILE_EXISTS_VALIDATOR)
+            .expressionLanguageSupported(true)
+            .build();
+
+    public static final PropertyDescriptor MORPHLINES_OUTPUT_FIELD = new PropertyDescriptor
+            .Builder().name("Morphlines output field")
+            .description("Field name of output in Morphlines. Default is '_attachment_body'.")
+            .required(false)
+            .defaultValue("_attachment_body")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final Relationship REL_SUCCESS = new Relationship.Builder()
+            .name("success")
+            .description("Relationship for success.")
+            .build();
+
+    public static final Relationship REL_FAILURE = new Relationship.Builder()
+            .name("failure")
+            .description("Relationship for failure of morphlines.")
+            .build();
+
+    public static final Relationship REL_ORIGINAL = new Relationship.Builder()
+            .name("original")
+            .description("Relationship for original flowfiles.")
+            .build();
+
+    private static final List<PropertyDescriptor> PROPERTIES = ImmutableList.<PropertyDescriptor>builder()
+            .add(MORPHLINES_FILE)
+            .add(MORPHLINES_ID)
+            .add(MORPHLINES_OUTPUT_FIELD)
+            .build();
+
+    private static final Set<Relationship> RELATIONSHIPS = ImmutableSet.<Relationship>builder()
+            .add(REL_SUCCESS)
+            .add(REL_FAILURE)
+            .build();
+
+    private File morphLineFile;
+    private String morphLinesId;
+    private String morphlinesOutputField;
+    private MorphlineContext morphlineContext;
```

Comment:
These should be made thread-safe (the onTrigger is almost certainly not going to be run by the same thread as the onScheduled).